src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeGenericParameterDesc.cs

```diff
@@ -25,5 +26,12 @@ public RuntimeGenericParameterDesc(GenericParameterKind kind, int index, TypeSys
         public override TypeSystemContext Context => _context;
 
         public override GenericVariance Variance => _variance;
+
+        public override TypeSystemEntity AssociatedTypeOrMethod => _associatedTypeOrMethod;
+
+        internal void SetAssociatedTypeOrMethod(TypeSystemEntity associatedTypeOrMethod)
+        {
+            _associatedTypeOrMethod = associatedTypeOrMethod;
+        }
```

Comment:
Can we do this in a constructor? This is already a `RuntimeXXXDesc` so I wouldn't mind if the constructor just understands a `ReadOnlySpan` of `Runtime.GenericVariance`.

Reply:
Refactor requested and implemented.